woocommerce/woocommerce-android
# Display variable subscription details

⚠️ This PR depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2684)

Closes: #8389 
Closes #8390 

### Description
This PR adds `read-only` support for variable subscriptions in the product details screen. It does that by reusing the variations' current flow within the product details screen and extending the functionality of the product variation class. 
Using the `SubscriptionDetailsMapper` created for simple subscriptions, we extract the subscription information from the variation metadata and display it to the merchants.

### Testing instructions

1. Open the Product List Screen
2. Tap on a variable subscription product
3. Check that the Variations / Variations attributes cards are shown
4. Tap on the Variations card
5. Check that all Subscription Variations are shown
6. Tap on the different Subscription Variations
7. Check that the subscription card is shown
8. Tap on the subscription card
9. Check that the app navigates to the subscription details screen and that the subscription details are shown

### Images/gif

https://user-images.githubusercontent.com/18119390/226401132-82a112a3-9543-4cc4-a829-58b3476d5a69.mp4
